### PR TITLE
Feature/update release checks

### PR DESCRIPTION
**Pull request by THOR300**
# Description

- Updates the publish workflow to check that the version in the latest release matches that which is declared in the version.py module of the cpr_sdk package. This means that we can't publish a package that states a certain version in pypi whilst the python code references another version. 

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [ ] Patch
- [x] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail (integrated soon)
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [x] Breaking change

## How Has This Been Tested?

- The command has been run locally.
- I've triggered the workflows and observed the expected failures due to version disparity. 

## Before submitting

- [x] I've read and followed all steps in the [Making a pull request](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [x] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [x] If this PR fixes a bug, I've added a test that will fail without my fix.
- [x] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

### CODE REVIEW

File: .github/actions/check-version-release/action.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    uses: actions/checkout@v4
+
+  - name: Setup Python
+    uses: actions/setup-python@v4
+    with:
+      python-version: "3.10"    
+
+  - shell: bash
+    run: |
+      pip install packaging==24.0
```
Comment (katybaulch) [inline .github/actions/check-version-release/action.yml:17]: This is a lot of Bash - we might want to think about make this a reusable workflow and moving this out of the repo so we can maintain it centrally

File: .github/actions/check-version/action.yml [hunk 2]
```diff
@@ -25,6 +25,7 @@ runs:
       
       # Check that semver tag is greater than the github release version
       PYPROJECT_GREATER_THAN_GITHUB_RELEASE=$(python -c "from packaging.version import Version; print(Version('${PYPROJECT_VERSION}') > Version('${GITHUB_RELEASE_VERSION}'))")
```
Comment (katybaulch) [inline .github/actions/check-version/action.yml:27]: Again, a lot of bash - we might want to think about make this a reusable workflow and moving this out of the repo so we can maintain it centrally

**Review by katybaulch — APPROVED** [review]
Nothing stopping this from going in, but a few pedantic comments and thoughts to leave you with 🦖

File: src/cpr_sdk/version.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
-_MAJOR = "0"
-_MINOR = "5"
-_PATCH = "9"
+_MAJOR = "1"
+_MINOR = "0"
+_PATCH = "1"
```
[the remark this answers is not in the document]
Comment (THOR300) [reply src/cpr_sdk/version.py:3]: Good spot, updated!

File: .github/actions/check-version-release/action.yml (hunk 1 above)
Comment (THOR300) [reply .github/actions/check-version-release/action.yml:17]: Yep absolutely agree!

File: .github/actions/check-version/action.yml (hunk 2 above)
Comment (THOR300) [reply .github/actions/check-version/action.yml:27]: Yep absolutely agree!

**Pull request merged**